platooning_strategic_IHP/src/platoon_manager_ihp.cpp

```diff
@@ -270,12 +272,18 @@ namespace platoon_strategic_ihp
     int PlatoonManager::allPredecessorFollowing(){
 
         ///***** Case Zero *****///
-        // If the host vehicle is the second vehicle in this platoon,we will always follow the platoon leader in front of host vehicle
-        if(hostPosInPlatoon_ == 1) {
-            ROS_DEBUG_STREAM("As the second vehicle in the platoon, it will always follow the leader. Case Zero");
-            return 0;
+        // If the host vehicle is the second follower of a platoon, it will always follow the platoon leader in the front 
+        if(getNumberOfVehicleInFront() == 1) {
+            if (isFollower){
+                // If the platoon length is 2 and the host is follower, then follow the leader.
+                ROS_DEBUG_STREAM("As the second vehicle in the platoon, it will always follow the leader. Case Zero");
+                return 0;
+            }
+            else{
```

Comment:
Continuing my rant, the else makes no sense.  Regardless of which branch we take, the return value is 0.  Why are we testing anything?  These are things to be ironed out after this PR.

Reply:
Yes, if this method is called, then this is a follower already. I think the entire case 0 is not necessary.